src/mame/drivers/minitel_2_rpic.cpp

```diff
@@ -0,0 +1,475 @@
+// license:BSD-3-Clause
+// copyright-holders: Jean-Francois DEL NERO
+/***************************************************************************
+
+	Minitel 2
+
+	The Minitel is a small, on-line computer/Videotex terminal with multi-services that
+	can be connected to any French telephone line. This terminal was widely used in France
+	during the 80's and 90's.
+
+	There are several modeles and version. Most of them are based on a mcu from the 8051 family
+	and a EF9345 like semi graphic video chip.
+
+	The current implementation is an Minitel 2 from "La RADIOTECHNIQUE PORTENSEIGNE" / RPIC (Philips)
+	You can found more informations about this hardware there :
+	http://hxc2001.free.fr/minitel
+
+	What is implemented and working :
+
+	- Main MCU
+	- Video output
+	- Keyboard
+
+	What is not yet implemented :
+
+	- Modem and sound output.
+	- The rear serial port.
+	- Parameters I2C 24C02 EEPROM.
+
+	The original firmware and the experimental demo rom are currently both working.
+
+	Please note the current special function keys assignation :
+
+	F1 -> Suite
+	F2 -> Retour
+	F3 -> Envoi
+	F4 -> Repetition
+	F5 -> TEL
+	F6 -> Guide
+	F7 -> Sommaire
+	F8 -> Connexion/Fin
+	F9 -> Fonction
+	F10-> ON / OFF
+
+	With the official ROM you need to press F10 to switch on the CRT.
+
+****************************************************************************/
+
+#include "emu.h"
+
+#include "cpu/mcs51/mcs51.h"
+#include "video/ef9345.h"
+
+#include "screen.h"
+#include "softlist.h"
+
+// IO expander latch usage definitions
+#define CTRL_REG_DTMF  0x02
```

Comment:
Please use `enum` or `constexpr` for numeric constants rather than preprocessor macros (`constexpr` variables may also be placed in an anonymous namespace).